sstmac/hardware/network/network_message.cc

```diff
@@ -285,6 +285,18 @@ void
 NetworkMessage::serialize_order(serializer& ser)
 {
   Flow::serialize_order(ser);
+
+  // can't serialize a nullptr!
+  // (and for UNPACK wire_buffer_ won't have been initialized yet)
+  if (ser.mode() == SST::Core::Serialization::serializer::SIZER ||
+      ser.mode() == SST::Core::Serialization::serializer::PACK ) {
+      if (wire_buffer_ == nullptr) {
+          wire_is_null_ = true;
+        }
+    }
+  ser & wire_is_null_;
```

Comment:
Seems like this check could just be, you can choose how much commenting is appropriate. 
```c++
bool wire_is_null = wire_buffer_ == nullptr;

// Unconditionally serialize wire_is_null, If SIZER or PACK this will write what ever our state is 
// If UNPACK this will overwrite our state
ser & wire_is_null;
if(!wire_is_null){ // If packing out wire is not null, if unpacking in wire is not null
  ser & sstmac::array(wire_buffer_, payload_bytes_);
} else { // If out wire is null don't write, if in wire is null don't read
  ser & payload_bytes_; 
}
```

Reply:
Resolved in CS 17bdc50